src/main/java/org/reaktivity/rym/internal/commands/install/Install.java

```diff
@@ -0,0 +1,193 @@
+            return false;
+        }
+        catch (IOException e)
+        {
+            // TODO Handle this error
+            System.err.format("Error: Something went wrong reading %s", DEFAULT_DEPENDENCY_FILENAME);
+            e.printStackTrace();
+            return false;
+        }
+        catch (JsonSyntaxException e)
+        {
+            // TODO Handle issue with JSON syntax
+            System.err.format("Error: %s is not in JSON format:", DEFAULT_DEPENDENCY_FILENAME);
+            System.err.println(e.getCause().getMessage());
+            return false;
+        }
+        // It's possible for a file to pass the parse phase, but still not be valid JSON. So do an explicit check.
+        if (!deps.isJsonObject())
+        {
+            // TODO Handle file not found
+            System.err.println(String.format("Error: %s is not in JSON format\n", DEFAULT_DEPENDENCY_FILENAME));
+            return false;
+        }
+
+        JsonObject depsObj = (JsonObject)deps;
+
+        JsonElement repositoriesEl = depsObj.get(DEPS_PROP_REPOSITORIES);
+        if (repositoriesEl != null)
+        {
+            if (!repositoriesEl.isJsonArray())
+            {
+                // TODO Handle file not found
+                System.err.format("Error: %s is not an array of strings", DEPS_PROP_REPOSITORIES);
+                return false;
+            }
+            JsonArray repositoriesArr = (JsonArray)repositoriesEl;
+            for (int i = 0; i < repositoriesArr.size(); i++)
+            {
+                if (repositoriesArr.get(i).isJsonPrimitive())
+                {
+                    repositories.add(repositoriesArr.get(i).getAsString());
+                }
+                else
+                {
+                    repositories.add(repositoriesArr.get(i).toString());
+                }
+            }
+        }
+
+        JsonElement dependenciesEl = depsObj.get(DEPS_PROP_DEPENDENCIES);
+        if (dependenciesEl != null)
+        {
+            if (!dependenciesEl.isJsonObject())
+            {
+                // TODO Handle file not found
+                System.err.format("Error: %s is not a JSON object", DEPS_PROP_DEPENDENCIES);
+                return false;
+            }
+            JsonObject dependenciesObj = (JsonObject)dependenciesEl;
+            dependenciesObj.keySet().forEach(key ->
+            {
+                String artifact = key.indexOf(':') > 0 ? key : String.format("%s:%s", DEFAULT_GROUP_NAME, key);
+                JsonElement versionEl = dependenciesObj.get(key);
+                // TODO Process version here. e.g. Is it a range?
+                if (versionEl.isJsonPrimitive())
+                {
+                    dependencies.put(artifact, versionEl.getAsString());
+                }
+                else
+                {
+                    dependencies.put(artifact, versionEl.toString());
+                }
```

Comment:
Try to avoid `if-else` clauses where the branches are almost identical. Instead extract the common part.

For example:
```
String version = versionEl.isJsonPrimitive() ? versionEl.getAsString() : versionEl.toString();
dependencies.put(artifact, version);
```

Now, this raises the question of whether or not the following simpler code would also always work.
```
String version = versionEl.getAsString();
dependencies.put(artifact, version);
```

Reply:
Looks like `getAsString()` only works on primitive types.